keps/sig-architecture/20190731-production-readiness-review-process.md

```diff
@@ -0,0 +1,258 @@
+merging as alpha or graduating to beta or GA. The review process is intended to
+ensure that features merging into Kubernetes are supportable, can be safely
+operated in production environments, and can be disabled or rolled back in the
+event they cause increased failures in production.
+
+## Motivation
+
+Kubernetes has grown quickly and organically. The KEP process introduced a
+mechanism to help build consensus and ensure the community supports the
+direction and implementation of various features. It provides a way to document
+release and graduation criteria, but it leaves that criteria up to the KEP
+authors and approvers to define. Because these are normally the developers and
+the SIGs associated with the KEP, there is not always a clear representation of
+the view of cluster operators. This can result in operational and supportability
+oversights in the graduation criteria.
+
+This KEP proposes a process to ensure that production concerns are addressed in
+all new features, at a level appropriate to the features' maturity levels.
+
+* TBD: Document some experiences that motivate the need for this.
+
+### Goals
+
+* Define production readiness criteria for alpha, beta, and GA features.
+* Define a production readiness review gate and process for all features.
+* Utilize existing tooling with prow to enforce the process.
+
+### Non-Goals
+
+* Building new tooling to enforce the process.
+
+## Proposal
+
+* Document production readiness criteria in a process document in the
+  kubernetes/community repository. Different levels of readiness may be
+  specified for different feature maturity levels.
+
+* Develop a production readiness questionnaire to ensure that the feature
+  authors consider and document operational aspects of the feature. Example
+  items:
+  * Feature enablement
+    - How can this feature be disabled in a live cluster? Will that require
+      downtime for the control plane?
+  * Dependencies
+    - Does this feature depend on any specific services running in the cluster
+      (e.g., a metrics service)?
+    - How does this feature respond to complete failures of the services on
+      which it depends?
+    - How does this feature respond to degraded performance or high error rates
+      from services on which it depends?
+  * Monitoring requirements
+    - How can an operator determine if the feature is in use by workloads?
+    - How can an operator determine if the feature is functioning properly?
+    - What are the service level indicators an operator can use to determine the
+      health of the service?
+    - What are reasonable service level objectives for the feature?
+  * Debugging
+  * etc.
+
+* Establish a production readiness review team, label, and CI check to prevent
+  the merging of feature promotion PRs that lack production readiness.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate.
```

Comment:
Delete this section?  Or may be add Risks of not moving forward with this proposal?